lib/yaml_db/serialization_helper.rb

```diff
@@ -57,16 +57,21 @@ def reenable_logger
 
     class Load
       def self.load(io, truncate = true)
-        ActiveRecord::Base.connection.transaction do
+        connection.transaction do
           load_documents(io, truncate)
         end
       end
 
       def self.truncate_table(table)
         begin
-          ActiveRecord::Base.connection.execute("TRUNCATE #{Utils.quote_table(table)}")
+          if connection.adapter_name == "SQLite"
+            connection.execute("DELETE FROM #{Utils.quote_table(table)}")
+            connection.execute("VACUUM")
```

Comment:
DELETE and VACUUM are the suggested queries to have the same behavior as TRUNCATE on other databases. I can't see any performance issues. We are already using this fix within a backup and restore functionality.